### PR TITLE
[IMPAC-768] - Allow user to schedule a transaction

### PR DESCRIPTION
**Pull request by agranado2k**
* Enable schedule section
* Allow user to repeat transactions as calendar event

### CODE REVIEW

**Comment by cesar-tonnoir** [comment]
@agranado2k There are a few conflict + the build is red - can you please have a look?
thanks

File: src/components/common/transactions-add/transactions-add.component.coffee [hunk 1]
```diff
@@ -1,31 +1,78 @@
...
+
+      ctrl.trx.datePicker =
+        opened: false
+        date: new Date()
+        toggle: -> this.opened = !this.opened
+
+      ctrl.schedulable =
+        recurring: false
+        interval: 4
+        rule: ctrl.schedulableRules[2]
```
Comment (xaun) [inline src/components/common/transactions-add/transactions-add.component.coffee]: Might be best to do `ctrl.schedulableRules.find((rule) -> rule.value == 'WeeklyRule')`, if a developer changes the order of that constant for some reason then this is broken. Also why is the `rule.value` in that capitalised camel case syntax and not something like `'weeklyrule'` or `'weekly_rule'` ?

File: src/components/common/transactions-add/transactions-add.component.coffee [hunk 2]
```diff
@@ -1,31 +1,78 @@
...
+        endType: 'occurrencies'
+        occurrencies: 12
+        rules:  ctrl.schedulableRules
+        datePicker:
           opened: false
           date: new Date()
           toggle: -> this.opened = !this.opened
 
+      if(!ctrl.editable )
```
Comment (xaun) [inline src/components/common/transactions-add/transactions-add.component.coffee]: `unless ctrl.editable`

File: src/components/common/transactions-add/transactions-add.component.coffee [hunk 3]
```diff
@@ -1,31 +1,78 @@
...
+    ctrl.includeSchedulable = (trx) ->
+      trx.recurring = ctrl.schedulable.recurring
+      if trx.recurring
+        trx.recurring_pattern = {
+          rule_type: ctrl.schedulable.rule.value,
+          interval: ctrl.schedulable.interval,
+          occurrences: ctrl.schedulable.occurrencies
+        }
+        if ctrl.schedulable.endType == 'endDate'
+          trx.recurring_end_date = ctrl.schedulable.datePicker.date
```
Comment (xaun) [inline src/components/common/transactions-add/transactions-add.component.coffee]: I would prefer:
```coffeescript
trx.recurring = ctrl.schedulable.recurring
return unless trx.recurring
trx.recurring_pattern = { ... }
trx.recurring_end_date = ctrl.schedulable.datePicker.date if ctrl.schedulable.endType == 'endDate'

File: src/components/common/transactions-add/transactions-add.less [hunk 4]
```diff
@@ -6,7 +6,7 @@
     width: 30%;
     .form {
       padding: 0px 10px;
-      > label, .coming-soon > label {
+      > label, .schedulable-section > label {
```
Comment (xaun) [inline src/components/common/transactions-add/transactions-add.less]: Rules following a comma should be on a new line

**Comment by cesar-tonnoir** [comment]
Pending https://github.com/agranado2k/impac-angular/pull/1

**Pull request merged**